public/app/plugins/datasource/elasticsearch/QueryBuilder.test.ts

```diff
@@ -19,11 +19,11 @@ describe('ElasticQueryBuilder', () => {
   it('should clean settings from null values', () => {
     const query = builder.build({
       refId: 'A',
-      // The following `missing: null as any` is because previous versions of the DS where
+      // The following `missing: null as unknown as string` is because previous versions of the DS where
       // storing null in the query model when inputting an empty string,
       // which were then removed in the query builder.
       // The new version doesn't store empty strings at all. This tests ensures backward compatibility.
-      metrics: [{ type: 'avg', id: '0', settings: { missing: null as any, script: '1' } }],
+      metrics: [{ type: 'avg', id: '0', settings: { missing: null as unknown as string, script: '1' } }],
```

Comment:
I feel like this backwards compatibility should be reflected in the actual type of the builder and not in casting `null` as string. Unrelated with this PR but now that I see it I don't like it.

Reply:
https://github.com/grafana/grafana/issues/81825